### PR TITLE
 Add C implementation for `@stdlib/math/base/special/rad2deg`

### LINKED ISSUE #654: [RFC]: Add C implementation for `@stdlib/math/base/special/rad2deg`
### Description

This RFC proposes adding a C implementation for [`@stdlib/math/base/special/rad2deg`](https://github.com/stdlib-js/stdlib/tree/b43a75923e47775446d3b6d7219ee42e07f7ce63/lib/node_modules/%40stdlib/math/base/special/rad2deg).

```c
double stdlib_base_rad2deg( const double x );
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Resolves #654 .

## Checklist

- [x] update readme.md
- [x] include.gypi
- [x] binding.gyp
- [x] include/stdlib/math/base/special/
- [x] src
- [x] manifest.json
- [x] lib
- [x] examples
- [x] benchmark
- [x] test
* * *

@kgryte

### CODE REVIEW

**Comment by Pranavchiku** [comment]
All the tests and benchmarks passed, this PR can be reviewed.

File: lib/node_modules/@stdlib/math/base/special/rad2deg/README.md [hunk 1]
```diff
@@ -90,6 +90,102 @@ for ( i = 0; i < 100; i++ ) {
...
+### Examples
+
+```c
+#include "stdlib/math/base/special/rad2deg.h"
+#include "stdlib/constants/float64/pi.h"
+#include <stdlib.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <math.h>
+#include <inttypes.h>
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rad2deg/README.md]:
```suggestion
#include <stdlib.h>
#include <stdio.h>
```

Removing unused includes.

File: lib/node_modules/@stdlib/math/base/special/rad2deg/examples/c/example.c [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/special/rad2deg.h"
+#include "stdlib/constants/float64/pi.h"
+#include <stdlib.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <math.h>
+#include <inttypes.h>
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rad2deg/examples/c/example.c]:
```suggestion
#include <stdlib.h>
#include <stdio.h>
```

File: lib/node_modules/@stdlib/math/base/special/rad2deg/lib/native.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+var addon = require( './../src/addon.node' );
+
+
+// MAIN //
+
+/**
+* Converts an angle from radians to degrees.
+*
+* @param {number} x - angle in radians
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rad2deg/lib/native.js:32]:
```suggestion
* @private
* @param {number} x - angle in radians
```

File: lib/node_modules/@stdlib/math/base/special/rad2deg/src/addon.c [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/special/rad2deg.h"
+#include <node_api.h>
+#include <stdint.h>
+#include <stdbool.h>
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rad2deg/src/addon.c]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/rad2deg/src/main.c [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+#include "stdlib/math/base/special/rad2deg.h"
+
+static const double CONST_180_DIV_PI = 57.29577951308232;
+
+/**
+* Converts an angle from radians to degrees.
+*
+* @param x    input value (in radians).
+* @return	  output value (in degrees).
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rad2deg/src/main.c]:
```suggestion
* @param x    input value (in radians)
* @return	  output value (in degrees)
```

File: lib/node_modules/@stdlib/math/base/special/rad2deg/src/main.c [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+*
+* @example
+* double x = 3.141592653589793/2.0
+*
+* double out = stdlib_base_rad2deg( x );
+*/
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rad2deg/src/main.c:34]:
```suggestion
* // returns 90.0
*/
```

**Review by kgryte — CHANGES_REQUESTED** [review]

**Review by kgryte — APPROVED** [review]

**Comment by kgryte** [comment]
LGTM. Small change we need to make in the `manifest.json` file, but I'll take care of that on my end. Thanks, @Pranavchiku!

**Pull request merged**